src/sunmatrix/sparse/sunmatrix_sparse.c

```diff
@@ -299,31 +290,34 @@ SUNMatrix SUNSparseFromBandMatrix(SUNMatrix Ad, sunrealtype droptol,
 /* ----------------------------------------------------------------------------
  * Function to create a new CSR matrix from a CSC matrix.
  */
-int SUNSparseMatrix_ToCSR(const SUNMatrix A, SUNMatrix* Bout)
+SUNErrCode SUNSparseMatrix_ToCSR(const SUNMatrix A, SUNMatrix* Bout)
 {
-  if (A == NULL) { return (SUNMAT_ILL_INPUT); }
-  if (SM_SPARSETYPE_S(A) != CSC_MAT) { return (SUNMAT_ILL_INPUT); }
+  SUNFunctionBegin(A->sunctx);
+  SUNAssert(SUNMatGetID(A) == SUNMATRIX_SPARSE, SUN_ERR_ARG_WRONGTYPE);
+  SUNAssert(SM_SPARSETYPE_S(A) == CSC_MAT, SUN_ERR_ARG_OUTOFRANGE);
 
   *Bout = SUNSparseMatrix(SM_ROWS_S(A), SM_COLUMNS_S(A), SM_NNZ_S(A), CSR_MAT,
                           A->sunctx);
-  if (*Bout == NULL) { return (SUNMAT_MEM_FAIL); }
-
-  return format_convert(A, *Bout);
+  SUNCheckLastErr();
+  SUNCheckCall(format_convert(A, *Bout));
+  return SUN_SUCCESS;
 }
 
 /* ----------------------------------------------------------------------------
  * Function to create a new CSC matrix from a CSR matrix.
  */
-int SUNSparseMatrix_ToCSC(const SUNMatrix A, SUNMatrix* Bout)
+SUNErrCode SUNSparseMatrix_ToCSC(const SUNMatrix A, SUNMatrix* Bout)
 {
-  if (A == NULL) { return (SUNMAT_ILL_INPUT); }
-  if (SM_SPARSETYPE_S(A) != CSR_MAT) { return (SUNMAT_ILL_INPUT); }
+  SUNFunctionBegin(A->sunctx);
+  SUNAssert(SUNMatGetID(A) == SUNMATRIX_SPARSE, SUN_ERR_ARG_WRONGTYPE);
+  SUNAssert(SM_SPARSETYPE_S(A) == CSR_MAT, SUN_ERR_ARG_OUTOFRANGE);
```

Comment:
Like above, should be `SUN_ERR_ARG_INCOMPATIBLE` or `SUN_ERR_ARG_WRONGTYPE` instead? And, should this be an error or should we just copy `A` to `B`?